File: python/problem171.py
import time
class Test:
    def test(self,c,repeat):
        t1=time.time()
        for i in range(repeat):
            c.titleToNumber(s)
        t2=time.time()
        print('time=',t2-t1)


class Solution:
    def titleToNumber(self, s):
        """
        :type s: str
        :rtype: int
        """
        cnt=0
        dic={chr(i):(i-64) for i in range(65,91)} #EXTRA SPACE ADN TIME
        for i in range(len(s)):
            cnt+=pow(26,len(s)-i-1)*dic[s[i]]
        return cnt
        
#FROM https://leetcode.com/problems/excel-sheet-column-number/discuss/183353/Easy-python-beats-100       
class Solution2(object):
    def titleToNumber(self, s):
        """
        :type s: str
        :rtype: int
        """
        result = 0
        s=s[::-1]
        for idx, val in enumerate(s):
            result += 26**idx * (ord(val)-ord('A')+1) 
            
        return result
        
if __name__ == '__main__':
    s='FHSDGJKGHDFKHJODSHGIRTHGSDFIOGHDFKHGDFKBHNKLFDKLENH'
    c=Solution()
    c1=Solution2()
    t=Test()
    t.test(c,100000)
    t.test(c1,100000)

File: swift/problem171.swift
class Solution1 {
    func pow(_ power: Int) -> Int {
        var result = 1
        for _ in 0..<power {
            result *= 26
        }
        return result
    }

    func titleToNumber(_ s: String) -> Int {
        let s = s.unicodeScalars
        var power = s.count - 1
        var result = 0
        
        for char in s {
            result += (Int(char.value) - 64)*pow(power)
            power -= 1
        }
        return result
    }
}

class Solution2 {
    func titleToNumber(_ s: String) -> Int {
        let s = s.unicodeScalars
        var result = 0
        
        for char in s {
            result *= 26
            result += (Int(char.value) - 64)
        }
        return result
    }
}
